Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -1762,7 +1763,62 @@ namespace AzToolsFramework
             }
         };
 
-        message.Match(AZ::DocumentPropertyEditor::Nodes::Adapter::QueryKey, showKeyQueryDialog);
+        auto showQuerySubclassDialog =
```

Comment:
are there any common scenarios where this dialog is needed?  do any of our components do this?

Reply:
not yet exactly, but I agree that it's a good idea to factor this out. For now, I want to get this issue fixed ASAP, so I going to merge this as-is, and work to factor out a Dialog in a following PR.